lib/src/main/java/com/evervault/services/HttpHandler.java

```diff
@@ -160,6 +164,64 @@ public <T> T decrypt(String url, Object data, Class<T> valueType) throws HttpFai
         return new Gson().fromJson(response.body(), valueType);
     }
 
+    @Override
+    public TokenResult createClientSideDecryptToken(String url, String action, Object data, Instant expiry) throws HttpFailureException, IOException, InterruptedException {
+        long expiryInMillis = expiry.toEpochMilli();
+
+        var payload = new CreateDecryptTokenPayload(action, expiryInMillis, data);
+        var serializedData = new Gson().toJson(payload);
+
+        var uri = URI.create(url);
+        var finalAddress = uri.resolve("/client-side-tokens");
+
+        var authHeaderValue = this.buildAuthorizationHeaderValue();
+        var requestBuilder = HttpRequest.newBuilder()
+            .uri(finalAddress)
+            .setHeader("User-Agent", VERSION_PREFIX + 1.0)
+            .setHeader("Accept", JSON_CONTENT_TYPE)
+            .setHeader("Content-Type", JSON_CONTENT_TYPE)
+            .setHeader("Authorization", authHeaderValue)
+            .timeout(httpTimeout)
+            .POST(BodyPublishers.ofString(serializedData));
+        var request = requestBuilder.build();
+
+        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+        if (response.statusCode() != OK_HTTP_STATUS_CODE) {
```

Comment:
The create token api returns a 201 on a successful response so this would fail.
Ideally we'd allow any 2xx status code though.